compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/bufferize_to_allocation_pad.mlir

```diff
@@ -0,0 +1,127 @@
+// RUN: iree-opt --pass-pipeline='builtin.module(func.func(iree-amdaie-bufferize-to-allocation{memory-space=1 bufferize-level=0}))' --split-input-file %s | FileCheck %s --check-prefix=BUFFERIZE-LEVEL-0
+// RUN: iree-opt --pass-pipeline='builtin.module(func.func(iree-amdaie-bufferize-to-allocation{memory-space=2 bufferize-level=1}))' --split-input-file %s | FileCheck %s --check-prefix=BUFFERIZE-LEVEL-1
+
+func.func @matmul_static(%arg0 : tensor<8x16xi32>, %arg1 : tensor<16x8xi32>) -> tensor<8x8xi32> {
+  %c0 = arith.constant 0 : index
+  %c0_i32 = arith.constant 0 : i32
+  %5 = tensor.empty() : tensor<8x8xi32>
+  %6 = scf.forall (%iv0, %iv1) = (0, 0) to (8, 8) step (8, 8) shared_outs(%arg2 = %5) -> (tensor<8x8xi32>) {
+    %extracted_slice_0 = tensor.extract_slice %arg0[%iv0, 0] [8, 16] [1, 1] : tensor<8x16xi32> to tensor<8x16xi32>
+    %extracted_slice_2 = tensor.extract_slice %arg1[0, %iv1] [16, 8] [1, 1] : tensor<16x8xi32> to tensor<16x8xi32>
+    %11 = tensor.empty() : tensor<8x8xi32>
+    %extracted_slice_3 = tensor.extract_slice %11[%iv0, %iv1] [8, 8] [1, 1] : tensor<8x8xi32> to tensor<8x8xi32>
```

Comment:
extracted_slice_3 isn't used, extracted_slice_4 is a no-op? 

It's not super simple IR: maybe add a sentence at the start of funcs about what the func does, where it's from, and what is being tested (to help make it easier understand what is being tested).

Reply:
Yes, this is taken taken directly from the output of padding pass without canonicalization to mimic the transform dialect transformations. I also feel some of the test cases are tedious and we may need to cleanup the test cases in the following patch.